[Documentation] - Adding doc about login (#11)
watchmycar/core-application-api: doc/login.md

# Login

> To a user login in the application has many ways to do.

Reviewer: What about replacing it with this? 
```suggestion
There are many ways to log a user into the application
```

## Simple Login

> The user can register on the platform using an email and a password, he has to be owner of the email that they using to register.

Reviewer: Maybe this would suit as well?
```suggestion
The user can register on the platform using an email and a password. They should be the owner of the email they are using in order to register.
```

## Social Login (Google and Facebook)

The user can register using another system of authentication.

## Database

To use all that types of login, the user table has to be like:

- id
- email
- password (NULL)
- name
- FacebookID
- GoogleID

## Working with multiples ways to login

> Using this struture the user can do any ways to login on the platform and add all the ways that he wants to login. For example, if a user register with Facebook, after he login in the application, he can add a password to he's account soo he can login with email and password. And the same way if he register with email and password, he can connect the social network to the account.

Reviewer: ```suggestion
The user can log into the platform using different methods by using the structure above. As an example, if a user registers with Facebook, after logging into the application, he will be able to add a password to his account and log into the application with email and password. The same applies if the user registers with an email and password and wants to login with Facebook.
```
